src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -1144,11 +1144,15 @@ public TimeSpan ErrorEventsHarvestCycle
         public IEnumerable<string> IgnoreErrorClassesForAgentSettings { get; private set; }
         public IDictionary<string, IEnumerable<string>> IgnoreErrorMessagesForAgentSettings { get; private set; }
 
-        private IEnumerable<MatchRule> ParseExpectedStatusCodesString(string expectedStatusCodesString)
+        private IEnumerable<MatchRule> ParseExpectedStatusCodesArray(IEnumerable<string> expectedStatusCodeArray)
         {
             var expectedStatusCodes = new List<MatchRule>();
 
-            var expectedStatusCodeArray = expectedStatusCodesString.Split(StringSeparators.Comma, StringSplitOptions.RemoveEmptyEntries);
+            if(expectedStatusCodeArray == null)
+            {
+                return expectedStatusCodes;
+            }
+
             foreach (var singleCodeOrRange in expectedStatusCodeArray)
             {
                 var index = singleCodeOrRange.IndexOf(HyphenChar);
```

Comment:
line 1162:   can it just be `var upperBoundString = singleCodeOrRange.Substring(index+1).Trim()`